galsim/inclinedexponential.py

```diff
@@ -0,0 +1,131 @@
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""@file inclinedexponential.py
+
+InclinedExponential is a class representing an exponential profile inclined to the LOS.
+"""
+
+import galsim
+from galsim import GSObject
+from . import _galsim
+
+
+class InclinedExponential(GSObject):
+    """A class describing an inclined exponential profile.
+
+    The Inclined Exponential surface brightness profile is characterized by three properties: its
+    inclination angle 'i' (where 0 = face-on and pi/2 = edge-on), its scale radius, and its scale
+    height. The 3D light distribution function is:
+
+    I(R,z) = I_0 / (2h_s) * sech^2 (z/h_s) * exp(-R/R_s)
+
+    where z is the distance along the minor axis, R is the radial distance from the minor axis,
+    R_s is the scale radius of the disk, h_s is the scale height of the disk, and I_0 is the central
+    surface brightness of the face-on disk.
+
+    In this implementation, the profile is inclined along the y-axis. This means that it will likely
+    need to be rotated in most circumstances.
+
+    At present, this profile is not enabled for photon-shooting.
+
+    Initialization
+    --------------
+
+    @param inclination      The inclination angle, i
+    @param scale_radius     The scale radius of the exponential disk.  Typically given in arcsec.
+    @param scale_height     The scale height of the exponential disk.  Typically given in arcsec.
+    @param flux             The flux (in photons) of the profile. [default: 1]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+
+    Methods
+    -------
+
+    In addition to the usual GSObject methods, InclinedExponential has the following access methods:
+
+        >>> inclination = inclined_exponential_obj.getInclination()
+        >>> r0 = inclined_exponential_obj.getScaleRadius()
+        >>> h0 = inclined_exponential_obj.getScaleHeight()
+    """
+    _req_params = { "inclination" : float, "scale_radius" : float, "scale_height" : float }
+    _opt_params = { "flux" : float }
+    _takes_rng = False
+
+    def __init__(self, inclination, scale_radius, scale_height,
+                 flux=1., gsparams=None):
+
+        # Explicitly check for angle type, so we can give more informative error if eg. a float is passed
+        if not isinstance(inclination, galsim.Angle):
+            raise TypeError("Input inclination should be an Angle")
+
+        GSObject.__init__(self, _galsim.SBInclinedExponential(inclination, scale_radius, scale_height, flux,
+                                                              gsparams))
+        self._gsparams = gsparams
+
+    def getInclination(self):
+        """Return the inclination angle for this profile in radians.
```

Comment:
That implies you get a float value in radians, whereas really I believe you get a galsim.Angle instance (which you can use to get whatever units you want).